bijection-util/src/main/scala/com/twitter/bijection/twitter_util/JavaFutureToTwitterFutureConverter.scala

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.bijection.twitter_util
+
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.{ Future => JFuture }
+
+import com.twitter.util.{ Future, Promise, Try }
+
+import scala.annotation.tailrec
+
+/**
+ * Utility class for converting Java futures to Twitter's
+ *
+ * @param waitTimeMs Time spent sleeping by the thread converting java futures to
+ *                   twitter futures when there are no futures to convert in ms
+ */
+private[twitter_util] class JavaFutureToTwitterFutureConverter(waitTimeMs: Long = 1000L) {
```

Comment:
This seems overly complicated (and induces a lot of latency).  How about instead the default is that we use a thread per future? the base implementation would be something like:

```scala
import com.twitter.util.{FuturePool, Promise, Future}
import java.util.concurrent.{Future => JFuture}
import scala.util.control.NonFatal

class FuturePoolConverter(mayInterruptWhileRunning: Boolean) {
  def apply[A](jfuture: JFuture[A]): Future[A] = {
    val f = FuturePool.unboundedPool { jfuture.get() }
    val p = Promise.detached(f)
    p.setInterruptHandler { case NonFatal(e) =>
      if (p.detach()) {
        f.raise(e)
        jfuture.cancel(mayInterruptWhileRunning)
      }
    }
    p
  }
}
```

A more sophisticated converter could use a Timer explicitly, and do something similar to what the current implementation does.

```scala
import com.twitter.util.{FuturePool, Promise, Timer, Try, Throw, Future}
import java.util.concurrent.{Future => JFuture}
import scala.util.control.NonFatal

class FuturePoolConverter(timer: Timer, checkFrequency: Duration, mayInterruptWhileRunning: Boolean) {
  def apply[A](jfuture: JFuture[A]): Future[A] = {
    val p = Promise[A]()
    val task = timer.schedule(checkFrequency) { 
      if (jfuture.isDone) {
        p.updateIfEmpty(Try(jfuture.get()))
        task.cancel()
      }
    }
    p.setInterruptHandler { case NonFatal(e) =>
      task.cancel()
      p.updateIfEmpty(Throw(e))
    }
    p
  }
}
```

Thoughts?